File: services/bot/src/utils/getGps.ts
import gps from "../utils/gps.json" assert { type: "json" };

export function getGps(year: string | number): string[] {
  const key = String(year).trim();
  const data = gps as Record<string, string[]>;
  console.log(data[key]);
  return data[key] ?? [];
}

File: services/core/utils/get_gps.py
import fastf1
import json
from pathlib import Path

data = {}

base = Path(__file__).resolve().parent
target = base.parent.parent / "bot" / "src" / "utils" / "gps.json"

for year in range(2019, 2026):
    schedule = fastf1.get_event_schedule(year)
    data[year] = schedule["Country"].tolist()

with target.open("w", encoding="utf-8") as f:
    json.dump(data, f, ensure_ascii=False, indent=2)
